index.js

```diff
@@ -90,7 +90,7 @@ function setNestedProperty(object, property, value) {
         if (typeof property == "string" && property !== "") {
             var split = property.split(".");
             return split.reduce(function (obj, prop, idx) {
-                obj[prop] = obj[prop] || {};
+                obj[prop] = obj[prop] || (!(isNaN(prop)) ? {} : []);
```

Comment:
are you sure isNaN does what you expect in this case?

`isNaN` tests if a value is `NaN`, but doesn't test if it's not a number. I would think it's safer to test the type: `typeof prop !== "number"` can you double check?

Reply:
You are right! With method isNaN will not work correct, I need check this.
The code `typeof prop !== "number"` will never works, because all props will be string, see this case  `"1" === String`. 
A solution possible would be use the regex `/^[0-9]$/`, What do you think ?

Sorry for my english and my contribution format, this is the first time I contribute to a project! haha

Reply:
you're right of course they come as string ;) You may use `Number` directly to convert to a number instead of `parseInt`. Using `parseInt` is dangerous when not using the optional radix given that the radix defaults to octal base instead of decimal base (depending on the implementation apparently). It also stops converting when it finds a non digit character:

```
parseInt("09") => 0 (in some implementation, node 10 doesn't do that anymore :)
parseInt("10eer", 10) => 10
Number("10eer") => NaN
```